core/io/file_access_pack.cpp

```diff
@@ -447,7 +447,11 @@ void FileAccessPack::set_big_endian(bool p_big_endian) {
 }
 
 Error FileAccessPack::get_error() const {
-	if (eof) {
+	if (encryption_error) {
+		return ERR_UNAUTHORIZED;
```

Comment:
`ERR_UNAUTHORIZED` is not used in other encrypted file cases, so it's probably better to store and return error code from `open_and_parse`.